File: quant1x/data/schema/sector.py
# -*- coding: utf-8 -*-
# Copyright (c) Quant1X <wangfengxy@sina.cn>.
# Licensed under the MIT License.

from dataclasses import dataclass, field
from typing import List

@dataclass
class Sector:
    """板块信息结构体"""
    name: str = ""
    code: str = ""
    type: int = 0
    count: int = 0
    block: str = ""
    constituent_stocks: List[str] = field(default_factory=List)

File: quant1x/data/schema/sector.rs
// Copyright (c) Quant1X <wangfengxy@sina.cn>.
// Licensed under the MIT License.

/// 板块信息结构体
#[derive(Debug, Clone, Default)]
pub struct Sector {
    /// 板块名称
    pub name: String,
    /// 板块代码
    pub code: String,
    /// 板块类型
    pub sector_type: i32,
    /// 成分股数量
    pub count: i32,
    /// 板块标识
    pub block: String,
    /// 成分股列表
    pub constituent_stocks: Vec<String>,
}
